src/tests/unit/tests/DetailsView/components/left-nav/left-nav-link-builder.test.tsx

```diff
@@ -182,61 +182,104 @@ describe('LeftNavBuilder', () => {
+    describe('buildAutomatedChecksLinks', () => {
+        it('should build just automated checks assessment link', () => {
+            const startingIndexStub = 0;
+            const {
+                assessmentsStub,
+                testStatusStub,
+                requirementStubA,
+                requirementStubB,
+                narratorStatusStub,
+                expandedTest,
+            } = createAssessmentsStubs();
+            const assessmentStub = assessmentsStub[0];
+            const { expectedTestLink, expectedGettingStartedLink } = getExpectedAssessmentLinks(
+                assessmentStub,
+                0,
+                startingIndexStub,
+                narratorStatusStub,
+                expandedTest,
+                testStatusStub,
+            );
 
-            assessmentsDataStub = {
-                [assessmentStub1.key]: stepStatusStub,
-                [assessmentStub2.key]: stepStatusStub,
-            };
+            const expectedRequirementLinkA = getExpectedRequirementLink(
+                requirementStubA,
+                assessmentStub.visualizationType,
+                testStatusStub,
+            );
+            const expectedRequirementLinkB = getExpectedRequirementLink(
+                requirementStubB,
+                assessmentStub.visualizationType,
+                testStatusStub,
+            );
+
+            const expectedAutomatedChecksLinks = [
+                expectedGettingStartedLink,
+                expectedRequirementLinkA,
+                expectedRequirementLinkB,
+            ];
+            const testLink = testSubject.buildAutomatedChecksLinks(
+                deps,
+                assessmentProviderMock.object,
+                assessmentsDataStub,
+                0,
+                expandedTest,
+                onRightPanelContentSwitchMock.object,
+            );
+            expect(testLink).toMatchObject(expectedTestLink);
+            testLink.links.forEach((actualLink, index) => {
+                expect(actualLink).toMatchObject(expectedAutomatedChecksLinks[index]);
+            });
+        });
+    });
+
+    describe('buildMediumPassTestLinks', () => {
+        it('should build links for medium pass tests', () => {
+            const startingIndexStub = 1;
+            const { assessmentsStub, testStatusStub, requirementStubA, requirementStubB } =
+                createAssessmentsStubs();
 
-            assessmentProviderMock.setup(apm => apm.all()).returns(() => assessmentsStub);
+            const links = testSubject.buildMediumPassTestLinks(
+                deps,
+                assessmentProviderMock.object,
+                assessmentsDataStub,
+                startingIndexStub,
+                onRightPanelContentSwitchMock.object,
+            );
 
-            outcomeStatsFromManualTestStatusMock
-                .setup(mock => mock(stepStatusStub))
-                .returns(() => outcomeStatsStub);
+            for (
```

Comment:
nit: I wonder when we write test code like this, is it better instead to do snapshot testing instead. Thoughts?

Reply:
Oh that's a good point! In general, I find our snapshots to be barebones in places so I'm glad we have reasonably identifiable stubs in this one.